distributed/deploy/ssh.py

```diff
@@ -86,25 +86,37 @@ def __init__(
         import asyncssh  # import now to avoid adding to module startup time
 
         self.connection = await asyncssh.connect(self.address, **self.connect_options)
-        self.proc = await self.connection.create_process(
-            " ".join(
-                [
-                    'DASK_INTERNAL_INHERIT_CONFIG="%s"'
-                    % serialize_for_cli(dask.config.global_config),
-                    sys.executable,
-                    "-m",
-                    self.worker_module,
-                    self.scheduler,
-                    "--name",
-                    str(self.name),
-                ]
-                + cli_keywords(self.kwargs, cls=_Worker, cmd=self.worker_module)
-            )
+        cmd = " ".join(
+            [
+                'env DASK_INTERNAL_INHERIT_CONFIG="%s"'
+                % serialize_for_cli(dask.config.global_config),
+                sys.executable,
+                "-m",
+                self.worker_module,
+                self.scheduler,
+                "--name",
+                str(self.name),
+            ]
+            + cli_keywords(self.kwargs, cls=_Worker, cmd=self.worker_module)
         )
 
+        self.proc = await self.connection.create_process(cmd)
```

Comment:
Should have a timeout?

Reply:
There is no timeout parameter for asyncssh connection. Please see the note on `asyncssh.connect`:

```
.. note:: Unlike :func:`socket.create_connection`, asyncio calls
                 to create a connection do not support a `timeout`
                 parameter. However, asyncio calls can be wrapped in a
                 call to :func:`asyncio.wait_for` or :func:`asyncio.wait`
                 which takes a timeout and provides equivalent functionality.
```
Something like below ... Should this be implemented here?

```python
try:
    self.connection = yield from asyncio.wait_for(asyncssh.connect(self.address, **self.connect_options), timeout=timeout)
    # do stuff with conn
    conn.close()
except TimeoutError:
    # handle connection timeout
```

Reply:
Sorry, I don't know the right way to do it either, but would have thought a timeout would be a good idea here. Perhaps someone else has best practices on this.